newer versions of openssl have started to be compatible across minor versions
too, so this sanity check fails.  since we already handle compatibility with
openssl via SONAME checks, we don't need this openssl check at all.

http://marc.info/?l=openssl-dev&m=133176786215023&w=2

--- src/ne_socket.c
+++ src/ne_socket.c
@@ -1745,13 +1745,6 @@ int ne_sock_connect_ssl(ne_socket *sock,
 	return NE_SOCK_ERROR;
     }
 
-    /* If runtime library version differs from compile-time version
-     * number in major/minor/fix level, abort soon. */
-    if ((SSLeay() ^ OPENSSL_VERSION_NUMBER) & 0xFFFFF000) {
-        set_error(sock, _("SSL disabled due to library version mismatch"));
-        return NE_SOCK_ERROR;
-    }
-
     sock->ssl = ssl = SSL_new(ctx->ctx);
     if (!ssl) {
 	set_error(sock, _("Could not create SSL structure"));
